api/topo/topo.go

```diff
@@ -24,6 +30,25 @@ type ID string
 // NullID ...
 const NullID = ""
 
+// Attribute keys
+const (
+	Revision    = "revision"
+	Address     = "address"
+	Target      = "target"
+	Type        = "type"
+	Version     = "version"
+	Timeout     = "timeout"
+	Role        = "role"
+	Displayname = "displayname"
+	User        = "user"
```

Comment:
I am not sure we should we include these kind of information in device attributes like username and password? Perhaps I am understanding is wrong.

Reply:
@adibrastegarnia - I agree. I would also like to make onos-topo super generic but the fact of the matter is that users of onos-topo (specifically onos-config) expect this information. At this point, I thought this is the best compromise solution.

Reply:
@shadansari 
OK. makes sense.

Reply:
I think `revision` is not necessary for `onos-config` to have